MinecraftForge/MinecraftForge
# Add missing AT lines to allow registering custom game rule value types

This PR adds the missing lines to the access transformer file to allow modders to register custom game rule value types, which are required in the game rules registration method `GameRules.register`.

These methods are `GameRules.BooleanValue.create` & `GameRules.IntegerValue.create`.

Forge previously only added an AT line for the game rules register method, but without AT'ing the value type methods modders could not register new game rules, as the register method takes in a value type.

While modders could easily AT these methods, forge should really be AT'ing them since they are already AT'ing the register method.

---

This PR also introduces a basic test to validate that custom game rules can be registered.

<details>

This test registers 2 new game rules
- `custom_game_rule_test:custom_boolean_game_rule`
- `custom_game_rule_test:custom_integer_game_rule`

To validate that these game rules were registered correctly
- Create a new world and ensure they show up under the `Edit Game Rules` screen (`Create new world > More (tab) > Game Rules > Misc`)
- Try and change their values in game using the following commands
	_These game rules should come up as valid auto-completion values if registered._
- `/gamerule custom_game_rule_test:custom_boolean_game_rule <true|false>`
- `/gamerule custom_game_rule_test:custom_integer_game_rule <some integer>`

</details>
